src/core/localStorageService.ts

```diff
@@ -0,0 +1,31 @@
+export default class LocalStorageService<JSON> {
+  #key: string
+  #storage: Storage
+
+  constructor(key: string) {
+    this.#storage = window.localStorage
+    this.#key = key
+  }
+
+  get(): JSON | null {
+    const value = this.#storage.getItem(this.#key)
+    if (value === undefined || value === null || value === '') return null
+    return this.toTargetType(value)
+  }
```

Comment:
```suggestion
  get(): JSON | null {
    const value = this.#storage.getItem(this.#key)
    if (value === undefined || value === null || value === '') return null
    return JSON.parse(value)
  }
```
한번 더 함수로 랩핑한 이유가 있을까요??  지금 storage에서 가져온 데이터를 JSON.parse말고 다른 API?로 파싱할것같지는 않은데 어떠한 이유로 랩핑하셨는지 궁금합니다.

Reply:
현재로서는 JSON형태로만 쓰고 있어서 말씀하신대로 불필요한 wrapping이긴 합니다. 처음에는 값을 그대로 toString처리만 해도 되는 경우에는 JSON이 아닌 문자열과 원본데이터간의 형변환 가능성도 생각하고 작성을 하다가, 마지막에 가서 문자열 형변환 케이스를 제거해놓고는 리팩토링을 하지 않았었네요. step3때 수정해 보겠습니다.